src/main/java/emu/grasscutter/game/quest/exec/ExecAddSceneTag.java

```diff
@@ -0,0 +1,26 @@
+package emu.grasscutter.game.quest.exec;
+
+import emu.grasscutter.data.common.quest.SubQuestData.QuestExecParam;
+import emu.grasscutter.game.quest.GameQuest;
+import emu.grasscutter.game.quest.QuestValueExec;
+import emu.grasscutter.game.quest.enums.QuestExec;
+import emu.grasscutter.game.quest.handlers.QuestExecHandler;
+import emu.grasscutter.server.packet.send.PacketPlayerWorldSceneInfoListNotify;
+import emu.grasscutter.server.packet.send.PacketSceneDataNotify;
+import lombok.val;
+import java.util.HashSet;
+
+@QuestValueExec(QuestExec.QUEST_EXEC_ADD_SCENE_TAG)
+public class ExecAddSceneTag extends QuestExecHandler {
+    @Override
+    public boolean execute(GameQuest quest, QuestExecParam condition, String... paramStr) {
+        val sceneNumber = Integer.parseInt(paramStr[0]);
+        val sceneTagNumber = Integer.parseInt(paramStr[1]);
+        val tagSet = quest.getOwner().getSceneTags().getOrDefault(sceneNumber, new HashSet<>());
+        tagSet.add(sceneTagNumber);
+        quest.getOwner().getSceneTags().put(sceneNumber, tagSet);
+        quest.getOwner().sendPacket(new PacketSceneDataNotify(quest.getOwner()));
+        quest.getOwner().sendPacket(new PacketPlayerWorldSceneInfoListNotify(quest.getOwner()));
```

Comment:
it might be good to put these all into a method thats part of scene/player to add/remove sceneTags.
this way there is a central point managing these and sending the packets, which can then also be used by Scriptlib or activity logic.

Reply:
Done. 
Moved it to player.